Update and refactor against upstream dolphin

cscd98 opened the pull request:
**Discussions are hidden under commits!!** This happens when github comment is older than the commits which are re pushed occasionally for patches.

This is an attempt to rebase this on upstream dolphin. There is more than 10,000 commits since the last upstream merge so there is quite a lot of changes required to get this to work.

This PR is up to date with upstream as of Oct 9, 2025 (https://github.com/dolphin-emu/dolphin/pull/13996)

It boots up a selection of games that I've tried, runs pretty darn fast (except in debug mode).

Ubuntu was my main testing and development platform. Here is what has been tested:

**Ubuntu:**
- Vulkan (recommended option)
- OpenGL
- Audio
- Input (Keyboard and Xbox controller tested)

**Windows 11:**
- D3D11, Vulkan and OpenGL backends
- Audio
- Input (Keyboard/Mouse)

**Android**
- Now works. Vulkan + OpenGL tested.

**macOS**
- Should now be working courtesy @warmenhoven

**iOS & tvOS**
- Supported thanks courtesy @warmenhoven

**Not tested:**
- Real wii controller as I do not own one

**New Features**
- Tons of changes due to merging in upstream
- Has 2 new options: Use async audio and Defer EFB Copies to RAM

**Future TODO**
- D3D12 is in development, new feature so not needed for this PR

**Cloning instructions**
```
git clone git@github.com:cscd98/dolphin.git
git checkout merge-clean-new
cd dolphin
git -c submodule."Externals/Qt".update=none \
-c submodule."Externals/FFmpeg-bin".update=none \
-c submodule."Externals/libadrenotools".update=none \
submodule update --init --recursive \
&& git pull --recurse-submodules
```

**Ubuntu 25.10 using GCC 15.2**
```
mkdir build && cd build
cmake .. -DLIBRETRO=ON -DCMAKE_BUILD_TYPE=Release
make -j$(getconf _NPROCESSORS_ONLN)

Note: if using archlinux - uninstall old mbedtls and mbedtls2, then install mbedtls
```

**Windows 10/11**
```
cmake -B build/win64-msvc -DLIBRETRO=ON -DCMAKE_BUILD_TYPE=Release -G"Visual Studio 17 2022" -A x64
cmake --build . --target dolphin_libretro --config Release
```

**Windows 10/11 using ming**
```
cmake -B build/win64-ming -DLIBRETRO=ON -DCMAKE_BUILD_TYPE=Release -G Ninja
ninja
```

**Cross compile from Ubuntu to Windows**
```
export PKG_CONFIG_LIBDIR=/usr/x86_64-w64-mingw32/lib/pkgconfig
export PKG_CONFIG_PATH=/usr/x86_64-w64-mingw32/share/pkgconfig
cmake -B build/win64-ming -DLIBRETRO=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=../toolchain-mingw64.cmake -G Ninja
ninja
```

**Android**
```
cmake -B build/android-arm64-v8a \
  -DLIBRETRO=ON \
  -DCMAKE_BUILD_TYPE=Release \
  -DANDROID_ABI=arm64-v8a \
  -DANDROID_PLATFORM=android-26 \
  -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK_HOME/build/cmake/android.toolchain.cmake
cd build/android-arm64-v8a
make -j$(getconf _NPROCESSORS_ONLN)
```

Shameless plug: If you want to thank or continue support for this please consider a small donation.. https://github.com/sponsors/cscd98

zorn-v committed:
Fix some errors

zorn-v committed:
Fix compile errors

zorn-v committed:
Fix linker errrors

zorn-v committed:
Init input before boot core

zorn-v committed:
Input init after boot core

Dentomologist committed:
EnhancementsWidget: Unabbreviate function and member names

Dentomologist committed:
EnhancementsWidget: Fix display of default post-processing effect

Show "(off)" instead of "" when the default post-processing effect is
selected. This also indirectly fixes issues with keyboard navigation of
the post-processing effect combobox when the default is selected,
resolving https://bugs.dolphin-emu.org/issues/13863.

m_post_processing_effect was previously using the ConfigStringChoice
constructor that assumes the text and data of each option are identical.
This is true for all the other effects, but since "(off)" has the config
value of "" this assumption was failing for it, causing the combobox to
be blank.

JoshuaVandaele committed:
pugixml: Update to 1.15 and make it a submodule

JoshuaVandaele committed:
discord-rpc: Bump CMake minimum requirement

SuperSamus committed:
CheatSearchWidget: "Add to watch" to all selected items

Instead of just the right-clicked item.

SuperSamus committed:
CheatSearchWidget: New feature, writing a value to all selected addresses

SuperSamus committed:
MMU: Use templates for Read/Write functions

jordan-woyak committed:
WiimoteReal: Send reports with proper timing for theoretically better speaker data.

Dentomologist committed:
CheatSearchWidget: Rename member variable

Rename m_session to m_last_value_session to better reflect its meaning
and distinguish it from m_current_value_session which will be introduced
in another commit.

Dentomologist committed:
CheatSearchWidget: Add GetTableRowCount function

TryTwo committed:
Update submodule implot

jordan-woyak committed:
DolphinQt: Make GameConfigHighlighter better handle large files.

cscd98 committed:
Update libretro to latest version from RetroArch

matheuswillder committed:
GameINI: Add 30 FPS codes for multiple games

GameINI: Add 30 FPS codes to Mario Kart: Double Dash, Mario Kart Wii

TryTwo committed:
Cheats: Refresh active code after editing it.

Dentomologist committed:
GameList: Use List View's sorting for Grid View

Since Grid View doesn't have a header for users to change sorting
options with, use List View's sorting in Grid View too.

Tilka committed:
VideoSW: reuse Common::Vec2/3/4

Sintendo committed:
JitArm64_Integer: subfcx - Optimize a == 0

This case can be handled as a move. It also generates a constant carry
flag.

Before:
0x52800013   mov    w19, #0x0                 ; =0
0x6b1302b3   subs   w19, w21, w19

After:
0x2a1503f3   mov    w19, w21

Sintendo committed:
JitArm64_Integer: subfcx - Optimize b == 0

Equivalent to a negation, no need to materialize the zero.

Before:
0x52800015   mov    w21, #0x0                 ; =0
0x6b1802b6   subs   w22, w21, w24

After:
0x6b1803f6   negs   w22, w24

Sintendo committed:
JitArm64_Integer: subfcx - Subtract 12-bit constant

Eliminate an instruction by encoding the constant in the subtraction.

Before:
0x5280037a   mov    w26, #0x1b                ; =27
0x6b1a02bb   subs   w27, w21, w26
0x93407f78   sxtw   x24, w27

After:
0x71006ebb   subs   w27, w21, #0x1b
0x93407f7a   sxtw   x26, w27

Sintendo committed:
JitArm64_Integer: subfcx - Subtract shifted 12-bit constant

Eliminate an instruction by encoding the constant in the subtraction,
part two.

Before:
0x52900019   mov    w25, #0x8000              ; =32768
0x6b190359   subs   w25, w26, w25

After:
0x71402359   subs   w25, w26, #0x8, lsl #12   ; =0x8000

JMC47 committed:
Merge pull request #13725 from Sam-Belliveau/more-consistent-looping

Fade audio after an entire loop

Dentomologist committed:
BranchWatch: Extract shared function logic

IsolateWasOverwritten and IsolateNotOverwritten share the same basic
logic and have almost exactly the same code, with the only difference
being the comparison function used to keep or discard branches. To avoid
unnecessary code duplication and ensure that the functions stay in sync
after any future changes, create a helper function that takes the
comparison function as a parameter and have IsolateWasOverwritten and
IsolateNotOverwritten call that helper.

JMC47 committed:
Merge pull request #13839 from Tilka/videosw3

VideoSW: allow disabling the copy filter

LillyJadeKatrin committed:
Fixed Patch Allowlist Check

Previous changes to the patch allowlist format were not reflected in AchievementManager; this corrects that and re-enables patches and codes in hardcore mode.

dreamsyntax committed:
PPCSymbolDB: Fix callers not updating

Fixes regression from http://github.com/dolphin-emu/dolphin/pull/13821

Tilka committed:
VideoSW: fix comment

Tilka committed:
IOS/FS: add safety assert

This is to prevent someone (me) from accidentally deleting their entire
/tmp directory just by not initializing the config system correctly.

Tilka committed:
Merge pull request #13787 from jordan-woyak/game-config-highlighter-fix

DolphinQt: Make GameConfigHighlighter better handle large files.

Tilka committed:
Merge pull request #13846 from JoshuaVandaele/better-xcb

Qt: Better wayland detection to enforce xcb

Tilka committed:
Merge pull request #13859 from Tilka/root_path_assert

IOS/FS: add safety assert

Tilka committed:
Merge pull request #13838 from Tilka/videosw2

VideoSW: reuse Common::Vec2/3/4

jordan-woyak committed:
HW/VideoInterface: Don't affect aspect ratio with VBI Frequency Override

JMC47 committed:
Merge pull request #13860 from SuperSamus/vbi-height-fix

HW/VideoInterface: Don't affect aspect ratio with VBI Frequency Override

Tilka committed:
Merge pull request #13854 from dreamsyntax/fix-regression-callers-broken

PPCSymbolDB: Fix callers not updating

Tilka committed:
DolphinQt: fix two -Wunused-lambda-capture warnings

Tilka committed:
Merge pull request #13861 from Tilka/unused_lambda_captures

DolphinQt: fix two -Wunused-lambda-capture warnings

TryTwo committed:
Update ImGui to 1.92.2b

TryTwo committed:
Update code to work with new version of ImGui.

UpdateImGuiTexture now handles creating font textures and modifying them.

Pokechu22 committed:
Mention enable MMU in the DSI Exception message (Invalid read from ###/Invalid write to ###)

This has been possible for a long time now (see https://dolphin-emu.org/blog/2016/09/06/booting-the-final-gc-game/#side-effects-of-a-hardcode-rewrite), but it seems like people still aren't aware of it.

Geotale committed:
Improve Interpreted FMADDS Precision

Improves the accuracy of FMADDS and other single precision FMA operations
This is accomplished by using an error-free transformation
It also thoroughly explains the quirks and difficulty of these operations
This fixes Mario Strikers and is necessary for fully fixing 1080 Avalanche
For single precision inputs it should be equivalent to a 32-bit FMA

TryTwo committed:
Fix bugs related to AbstractStagingTextures that perform an Upload (write to existing texture).  This code path had probably never been used before.

JMC47 committed:
Merge pull request #13867 from TryTwo/gfx_bug_fix

GFX: Fix bugs for AbstractStagingTextures that perform an Upload.

Dentomologist committed:
DolphinAnalytics: Use Pascal case for GameQuirk enum

Tilka committed:
Jit64: drop unused constant

Unused since c4799e59771115f4d5050a0fd5fb299f140b411d.

Pokechu22 committed:
Merge pull request #13865 from Pokechu22/dsi-mention-mmu

Mention enable MMU in the DSI Exception message (Invalid read from ###/Invalid write to ###)

sepalani committed:
Common/BitUtils: Add operator[] to BitCastPtrType

iwubcode committed:
Data: add gecko code for Mario Strikers Charged that allows disabling captain mega strikes

iwubcode committed:
VideoBackends: allow custom pixel uniforms to be passed to the vertex shader as well

Tilka committed:
Merge pull request #13853 from LillyJadeKatrin/retroachievements-patch-fix

Fixed Patch Allowlist Check

Tilka committed:
OpcodeDecoder: provide default GetVertexSize() implementation

Tilka committed:
FIFOAnalyzer: fix command description updates

When dragging the selection, the mismatch between signal
(itemSelectionChanged) and data consumed (currentRow) seemed to cause
the description to lag behind by one row.

OatmealDome committed:
FloatUtils: Move MakeQuiet function here from Interpreter_FPUtils

OatmealDome committed:
FloatUtils: Replace quieting SNaNs via `0.0 + x` with MakeQuiet

OatmealDome committed:
Merge pull request #13881 from OatmealDome/clang-floating-point-nonsense-2

FloatUtils: Replace quieting SNaNs via `0.0 + x` with MakeQuiet

jordan-woyak committed:
ControllerInterface/SDL: Don't force SDL_HINT_JOYSTICK_THREAD.
SDL3 enables it by default now and things seem to work properly even when it's off these days.

jordan-woyak committed:
ControllerInterface/SDL: Remove manual window message pumping on Windows.
The mentioned SDL hotplug issue seems to have been fixed.

Dentomologist committed:
FIFOPlayerWindow: Add stretch to Play/Record tab

Dentomologist committed:
FIFOPlayerWindow: Trigger destructor on Dolphin shutdown

Make MainWindow::m_fifo_window a unique_ptr to ensure its destructor is
triggered when MainWindow is destroyed.

FIFOPlayerWindow doesn't set MainWindow as its parent in order to
prevent raising MainWindow when focusing FIFOPlayerWindow. This avoids
MainWindow covering up RenderWidget when, e.g., trying to use the object
range feature to pinpoint the index of a particular object.

As a consequence, unlike most QObjects FIFOPlayerWindow wasn't destroyed
when its parent widget was since it didn't have one.

Dentomologist committed:
FIFOPlayerWindow: Save and restore window geometry

Aside from allowing users to persistently set the window to their
desired size, this is also necessary to allow saving of the splitter
positions in FIFOAnalyzer to work correctly.

JMC47 committed:
Merge pull request #13879 from Tilka/get_vertex_size

OpcodeDecoder: provide default GetVertexSize() implementation

Tilka committed:
Merge pull request #13883 from Dentomologist/fifoplayerwindow_geometry_fixes

FIFOPlayerWindow geometry fixes

JMC47 committed:
Merge pull request #13868 from Geotale/master

Fix Single-Precision-Only Inputs to FMAs Instructions in Interpreter

Tilka committed:
Merge pull request #13887 from Tilka/unused_sign_bit

Jit64: drop unused constant

Tilka committed:
Merge pull request #13873 from Dentomologist/dolphinanalytics_use_pascal_case_for_gamequirk_enum

DolphinAnalytics: Use Pascal case for GameQuirk enum

JMC47 committed:
Merge pull request #13877 from iwubcode/vertex_shader_custom_uniforms

VideoBackends: allow custom pixel uniforms to be passed to the vertex shader as well

Sintendo committed:
VideoCommon: Remove unused EnumMaps from PixelShaderGen

They appear to have been unused since #13432.

JosJuice committed:
Merge pull request #13836 from Sintendo/jitarm64-subfe

JitArm64_Integer: Minor subfe optimizations

Tilka committed:
Merge pull request #13888 from Sintendo/unused-psg

VideoCommon: Remove unused EnumMaps from PixelShaderGen

iwubcode committed:
VideoCommon: add TextureAndSamplerAsset, this asset contains both the raw texture data and data about how the texture should be sampled

jordan-woyak committed:
GameINI: Disable ImmediateXFBEnable for Grooverider.

JosJuice committed:
Translation resources sync with Transifex

JMC47 committed:
Merge pull request #13891 from jordan-woyak/grooverider-ini

GameINI: Disable ImmediateXFBEnable for Grooverider.

JosJuice committed:
AchievementManager: Use HookableEvent instead of std::function callbacks

The callback mechanism AchievementManager had until now only supported
one caller registering a callback, and it didn't have any
synchronization. This isn't a problem for DolphinQt, but the PR to add
Android support for RetroAchievements exposes these problems. Let's
replace it with HookableEvent, which can handle all of this.

TryTwo committed:
Qt: Prevent Sliders from resizing due to text changing in their label.

JMC47 committed:
Merge pull request #13764 from SuperSamus/patch-invalidate-cache

ActionReplay/PatchEngine: Make writes invalidate cache

iwubcode committed:
VideoCommon: update material asset to support render state properties, support multi-pass, and textures are now split out (as well as supporting a way to calculate sampler origin)

JMC47 committed:
Merge pull request #13886 from Tilka/drag_lag

FIFOAnalyzer: fix command description updates

iwubcode committed:
VideoCommon: update shader asset to support both vertex and pixel shaders; rename to 'RasterSurfaceShader'

iwubcode committed:
VideoCommon: further remove 'CustomPipeline' details, so we can fully remove it in the very near future

JMC47 committed:
Merge pull request #13869 from TryTwo/ImGui_Updates

Externals: Update ImGui to 1.92.2 and ImPlot. Update code to use new features.

Tilka committed:
VideoCommon: fix -Wmismatched-tags warning

Tilka committed:
Merge pull request #13896 from Tilka/mismatched_tags

VideoCommon: fix -Wmismatched-tags warning

SuperSamus committed:
ActionReplay: Fixup #13764

Tilka committed:
Merge pull request #13898 from SuperSamus/patch-invalidate-cache-fixup

ActionReplay: Fixup #13764

Tilka committed:
Merge pull request #13893 from TryTwo/PR_Slider_Fixes

Qt: Prevent QSlider's QLabels from resizing.

SuperSamus committed:
PatchEngine: OSD message showing number of enabled patches and cheats

iwubcode committed:
VideoBackends / VideoCommon: rename member variables in RenderState to be consistent

Tilka committed:
Merge pull request #13901 from iwubcode/render_state_rename

VideoBackends / VideoCommon: rename member variables in RenderState

Tilka committed:
VideoSW: fix clamping after vertex color interpolation

JMC47 committed:
Merge pull request #13894 from iwubcode/material_asset_update

VideoCommon: update material asset to support additional features in custom pipelines

JosJuice committed:
Arm64Emitter: Replace shifting size by 4 with IntLog2 minus 3

The instruction implementations that were shifting the size by 4 would
emit an incorrect instruction when given a size of 64. The correct
implementation is to count the number of leading or trailing zeroes in
the size parameter, which is what IntLog2 does.

No callers are affected by this, as they all use sizes other than 64.
Actually, some of these instructions are even invalid with a size of 64,
but I'm changing them anyway for consistency with the others.

Dentomologist committed:
CodeWidgets: Show code approval in Hardcore mode

When Hardcore mode is enabled, show an icon for each code in
ARCodeWidget and GeckoCodeWidget indicating whether it's an approved
code or not.

Marocco2 committed:
Android: Set Dolphin documents as Local only

Adds `DocumentsContract.Root.FLAG_LOCAL_ONLY` to the list of the flags in order to show up for third-party apps for easier file syncing with local/cloud file server providers

cristian64 committed:
DolphinQt: Center dialogs on top-level parent window.

The `QtUtils::AdjustSizeWithinScreen()` function now centers the
widget on their parent window after the widget's size is adjusted. This
is required in some desktop environments (generally on Linux systems) to
ensure that the just-resized widget remains aligned with its parent
window.

The creation of the **Cheats Manager** dialog is now deferred to first
show, as creating it within the main window's constructor prevents the
dialog from knowing the real position of its parent window, which is
only properly calculated on first show.

For the same reasons, the analytics prompt is now shown only when the
application is ready (i.e. when the main window has been shown).

| Before | After |
| ------ | ----- |
| <img width="1920" height="1080" alt="[Dolphin Emulator] Misplaced analytics prompt" title="[Dolphin Emulator] Misplaced analytics prompt" src="https://github.com/user-attachments/assets/c3e43b3b-cf79-4398-b531-7de6068c583e" /> | <img width="1920" height="1080" alt="[Dolphin Emulator] Centered analytics prompt" title="[Dolphin Emulator] Centered analytics prompt" src="https://github.com/user-attachments/assets/a10d06d9-7438-4032-b96c-dfcb48826349" /> |
| <img width="1920" height="1080" alt="[Dolphin Emulator] Misplaced Settings dialog" title="[Dolphin Emulator] Misplaced Settings dialog" src="https://github.com/user-attachments/assets/f35b10ee-4f07-48d4-86f9-2537ad5ca7ca" /> | <img width="1920" height="1080" alt="[Dolphin Emulator] Centered Settings dialog" title="[Dolphin Emulator] Centered Settings dialog" src="https://github.com/user-attachments/assets/33e37237-77a4-44f6-a0e2-b709f65b672b" /> |
| <img width="1920" height="1080" alt="[Dolphin Emulator] Misplaced Cheats Manager dialog" title="[Dolphin Emulator] Misplaced Cheats Manager dialog" src="https://github.com/user-attachments/assets/1fbd3836-5639-4d5a-b57e-e2e21f21c9db" /> | <img width="1920" height="1080" alt="[Dolphin Emulator] Centered Cheats Manager dialog" title="[Dolphin Emulator] Centered Cheats Manager dialog" src="https://github.com/user-attachments/assets/3f2b1b50-de16-49b7-bac4-c444c6cab0bc" /> |

Tilka committed:
IOS/WD: fix wrong BitSet ctor being called

We don't want list initialization here.

Thank you, UB-san.

Tilka committed:
VI: set default overclock factor to 1.0

GetTicksPerHalfLine() gets called via Preset() before RefreshConfig()
has a chance to replace the dummy value 0.0.

Thank you, UB-san.

jordan-woyak committed:
MathUtil: Simplify SaturatingCast implementation with std::cmp_less/cmp_greater and fix a floating point edge case.
Thanks to Dentomologist for catching the edge case.

dreamsyntax committed:
Sys: Sync wiitdb files with GameTDB

Sync with 20250826093514

dreamsyntax committed:
Externals: Update discord-rpc

Upgrade to 963aa9f3e5ce81a4682c6ca3d136cddda614db33

Marocco2 committed:
Android: Set Dolphin documents' root supports testing parent child relationships.

Adds `DocumentsContract.Root.FLAG_SUPPORTS_IS_CHILD` to the list of the flags in order to show up for third-party apps for easier file syncing with local/cloud file server providers

JMC47 committed:
Merge pull request #13913 from dreamsyntax/titledb-update

Sys: Sync wiitdb files with GameTDB 2025.08

JMC47 committed:
Merge pull request #13825 from jordan-woyak/dont-count-playtime-while-suspended

Common/Timer: Add a SteadyAwakeClock class to make play time tracking ignore time while suspended.

JordanTheToaster committed:
Externals: Update gtest to commit eb2d85edd0bff7a712b6aff147cd9f789f0d7d0b

Tilka committed:
Merge pull request #13911 from Dentomologist/gamelist_gridview_sorting

GameList: Use List View's sorting for Grid View

Tilka committed:
Merge pull request #13912 from jordan-woyak/simplify-saturating-cast

MathUtil: Simplify SaturatingCast implementation and fix edge case.

Tilka committed:
Merge pull request #13902 from Tilka/fix_color_clamp

VideoSW: fix clamping after vertex color interpolation

Tilka committed:
CoreTiming: set default overclock factors to 1.0

I'm not aware of any issue here but let's set meaningful defaults
anyway.

Tilka committed:
Merge pull request #13910 from Tilka/ub

Fix some undefined behavior

Tilka committed:
Merge pull request #13899 from SuperSamus/patch-cheats-osd

PatchEngine: OSD message showing number of enabled patches and cheats

AdmiralCurtiss committed:
DolphinQt/Styles/Dark: Fix inconsistent formatting

AdmiralCurtiss committed:
DolphinQt/Styles/Dark: Adjust progress bar

Dentomologist committed:
Settings: Open file dialogs at user path

When opening a file dialog to set the location of a custom path, use the
corresponding user path as the starting location instead of the current
custom path.

When no custom path was set the dialog would be opened with a blank path
which causes Windows (not sure about other platforms) to open the dialog
at the same location where the last dialog was closed, or at the current
working directory if no previous dialog had been opened.

If a nonempty custom path has been set then it has also set the
corresponding user path, so the behavior in that case is unchanged.

SuperSamus committed:
x64Emitter: Support YMM registers

This is accomplished by adding a 0x100 bit the the register.
Made sure that, on AVX instructions, that bit is trucated.

SuperSamus committed:
Jit64: `dcbz`, use VMOVAPS with YMM registers on AVX CPUs
LLVM does this, so let's do it as well.

SuperSamus committed:
Jit: Use `dcbz()` emitter for `dcbz_l` instead of interpreter fallback
Without cache emulation, these instructions are functionally identical.
In the interpreter, their only difference is related to HID registers checks, which the JIT already doesn't do for `dcbz`.
A loop with `dcbz_l` is used in the SDK function `LCEnable`, which is called frequently in some games.

JosJuice committed:
Translation resources sync with Transifex

cscd98 committed:
Refactor DolphinLibretro on latest upstream changes

cscd98 committed:
Refactor base code for libretro usage

cscd98 committed:
Re-add RetroArch to CMakeLists

Dentomologist committed:
RetroAchievements: Avoid uninitialized memory access

Avoid creating and then destroying a leaderboard list when game is null,
as doing so causes an access to uninitialized memory due to a bug in
rcheevos.

This can be triggered by starting a game with an invalid or expired
login token.

TellowKrinkle committed:
Cached Interpreter: Speculative devirtualization

apteryks commented:
I was able to build it on Guix! The libretro include not found error earlier was caused by a packaging bug on my side.

I could run a couple games, but the sound is often choppy, and the performance is down (from a solid 60 fps in wii games to e.g. 40-50, with choppy sound), so perhaps something needs to be optimized/fixed.

Thanks again for doing this!

1985a commented:
Finally, I was able to compile this core. I want to say, MANY thanks for making this. I was waiting for many months for this core to be updated.

Now I came across this issue. 
<img width="961" height="231" alt="image" src="https://github.com/user-attachments/assets/48475525-f452-46e7-8d10-de7b172a212b" />

AdmiralCurtiss committed:
Merge pull request #13923 from AdmiralCurtiss/dark-progress-bar

DolphinQt/Styles/Dark: Adjust progress bar

JoshuaVandaele committed:
RealtekFirmwareLoader: Fix shadowed variable warning

jordan-woyak committed:
Merge pull request #13933 from JoshuaVandaele/rt-firmware-shadow

RealtekFirmwareLoader: Fix shadowed variable warning

iwubcode committed:
Core: support inverting a 4x4 matrix

apteryks commented:
@cscd98 hello! I tried the latest commit, and the choppy audio problem remains; it seems the async ON makes it worse compared to async audio OFF setting.

apteryks commented:
> @cscd98 hello! I tried the latest commit, and the choppy audio problem remains; it seems the async ON makes it worse compared to async audio OFF setting.

I was curious to compare apples to apples, so I've tested using the standalone `dolphin-emu` at version 2506-0.79a3af2 (as packaged in Guix), and to my surprise, its performance is similar. So it seems the performance regression, at least with this game, probably has more to do with the newer dolphin-emu code itself than your port.

apteryks commented:
More comparison data: I just time travelled back to dolphin-emu 5.0 via `guix time-machine -q --commit=81a956d5fa4 -- shell coreutils qtwayland dolphin-emu -- env QT_QPA_PLATFORM=xcb dolphin-emu`, and indeed the performance was much better there, for this game at least.

TryTwo committed:
Fix broken imgui text in Metal backend.

jordan-woyak committed:
Merge pull request #13936 from TryTwo/imgui_bugfix

ImGui: Fix broken imgui text in Metal backend.

jordan-woyak committed:
DolphinQt: Clean up some of the style sheet hacks in StackedSettingsWindow.

jordan-woyak committed:
GameINI: Add 60 FPS patch for NTSC version of Terminator 3.

iwubcode committed:
Externals / VideoCommon: update glslang to latest, turn into submodule

JMC47 committed:
Merge pull request #13930 from Dentomologist/retroachievements_avoid_leaderboard_list_crash

RetroAchievements: Avoid crash due to uninitialized memory read

jordan-woyak committed:
VideoCommon/OpcodeDecoding: Replace enable_if with concepts.

1985a commented:
> Thank you @cscd98 and @1985a
> 
> I confirm it is working on Linux using
> 
> `cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS="-march=native -O2 -flto -DNDEBUG" -DCMAKE_CXX_FLAGS="-march=native -O2 -flto -DNDEBUG" -DCMAKE_POLICY_VERSION_MINIMUM=3.5 -DLIBRETRO=ON -DENABLE_ASSERTS=OFF`
> 
> Even if it says
> 
> ```
> CMake Warning:
>   Manually-specified variables were not used by the project:
> 
>     ENABLE_ASSERTS
> ```
> 
> But I reproduce the input controller issue
> 
> > Now I'm facing only one problem, apparently the core does not see the input controller, but retroarch shows me the trigger on the overlay.
> 
> I tried to change Settings > Input > Polling behavior but it did not help.

I think those commits introduce the problem
<img width="389" height="52" alt="image" src="https://github.com/user-attachments/assets/ae18b4a4-8435-4aca-9d51-9c16fef04f14" />


They were added on sep 12

jordan-woyak committed:
Merge tag '2506a' into release-prep-2509

First hotfix release for 2506

jordan-woyak committed:
ScmRevGen: Bump version to 2509

jordan-woyak committed:
ScmRevGen: Bump version to 2509

1985a commented:
> @1985a What controller are you using? Is it USB/Bluetooth?

xbox wireless, but retroarch seens the controller, the overlay shows the input trigger and this core was working well when i first started to test it. 


I dont see how to go back to specific commit to compile cause the libretro core dont want to compile if i do that.

JosJuice committed:
Merge pull request #13905 from JosJuice/arm64emitter-shift-4

Arm64Emitter: Replace shifting size by 4 with IntLog2 minus 3

JosJuice committed:
Merge pull request #13908 from Marocco2/patch-1

Android: Set Dolphin documents as Local only

JosJuice committed:
Merge pull request #13852 from Sintendo/jitarm64-subfcx

JitArm64_Integer: subfcx optimizations

JosJuice committed:
Merge pull request #13847 from jordan-woyak/sdl-version-bump

Externals/SDL: Update submodule to release 3.2.20.

JMC47 committed:
Merge pull request #13682 from JoshuaVandaele/libspng-cmake4

libspng: Update to 0.7.4 and clean up CMake

JMC47 committed:
Merge pull request #13863 from Dentomologist/enhancements_tab_fix_selection_of_disabled_postprocessing_option

EnhancementsWidget: Fix selection of disabled postprocessing option

JMC47 committed:
Merge pull request #13864 from Dentomologist/branchwatch_extract_shared_function_logic

BranchWatch: Extract shared function logic

JMC47 committed:
Merge pull request #13876 from iwubcode/strikers_charged_cheat

Data: add gecko code for Mario Strikers Charged that allows disabling captain mega strikes

JMC47 committed:
Merge pull request #13889 from iwubcode/texture_and_sampler_asset

VideoCommon: add TextureAndSamplerAsset for use in custom materials

JMC47 committed:
Merge pull request #13892 from JosJuice/retroachievements-hookableevent

AchievementManager: Use HookableEvent instead of std::function callbacks

JMC47 committed:
Merge pull request #13895 from iwubcode/update_shader_asset

VideoCommon: update shader asset to support additional features in custom pipelines

JMC47 committed:
Merge pull request #13917 from cristian64/center_dialogs

DolphinQt: Center dialogs on top-level parent window.

JMC47 committed:
Merge pull request #13934 from jordan-woyak/stacked-settings-style-cleanup

DolphinQt: Clean up some of the style sheet hacks in StackedSettingsWindow.

JMC47 committed:
Merge pull request #13935 from iwubcode/matrix_inverted

Core: support inverting a 4x4 matrix

JMC47 committed:
Merge pull request #13937 from jordan-woyak/terminator3-speedhack

GameINI: Add 60 FPS patch for NTSC version of Terminator 3.

JMC47 committed:
Merge pull request #13941 from jordan-woyak/opcode-concepts

VideoCommon/OpcodeDecoding: Replace enable_if with concepts.

JMC47 committed:
Merge pull request #13680 from JoshuaVandaele/enet-cmake4

enet: Update to 1.3.18 and clean up CMake

JMC47 committed:
Merge pull request #13848 from JoshuaVandaele/USBDevicePicker

USBDevicePicker: Modify USBDeviceAddToWhitelistDialog to be more generic, and use it for a new "More Options..." selection in Bluetooth Passthrough adapters

OatmealDome committed:
Merge branch 'release-prep-2509'

OatmealDome committed:
Merge tag '2509'

Release for September 2025

iwubcode committed:
Common: add transpose function to Matrix class

JMC47 committed:
Merge pull request #13874 from sepalani/bitcast-ptr

Common/BitUtils: Add operator[] to BitCastPtrType

cscd98 committed:
Fix cmake compilation on Windows 11

JosJuice committed:
i18n: Add comments

1985a commented:
I don't know why, I tested the drivers subject and it still nothing. I just left this, for another day. Thank you for your effort to update this core.

TryTwo committed:
OSD / Imgui: Add font size spinbox to Interface Pane.

Adjust some imgui windows to work better with different font sizes.

TryTwo committed:
Crash fix: Add check to see if debugging is enabled in Jit's IsProfilingEnabled().

Enable JIT Block Profiling = On, with Enable Debugging UI = Off, will crash dolphin when starting a game.

jordan-woyak committed:
Common: Add some utilities to a new UnixUtil header.

jordan-woyak committed:
WiimoteReal/IOLinux: Improvements, fixes, and code cleanups.

JMC47 committed:
Merge pull request #13795 from matheuswillder/30-FPS-code

GameINI: Add 30 FPS codes to Mario Kart: Double Dash, Mario Kart Wii

Xphalnos committed:
Externals: Update FMT to 12.0.0

jordan-woyak committed:
CommonFuncs: Add StrerrorString version of LastStrerrorString that accepts an error number.

gouchi commented:
I made a test on Wayland with the core using [a93180d](https://github.com/libretro/dolphin/commit/a93180d817d636034bd7dd0dab170013f7e63ac9) and I confirm it is working for both GC and Wii with udev controller driver !

srk15372 commented:
Hi great work, hope you could achieve the work and bring us a valid core on the buildbot
Regards

1985a commented:
Finally, the core responds to the inputs. Now it works.

jordan-woyak committed:
Common/Network: Add BluetoothAddress struct and string conversion functions.

jordan-woyak committed:
WiimoteEmu: Require IRPassthrough has any bound inputs to operate. Users get confused and mistakenly enable it.

ryanmcgrath committed:
macoS: update QoSSession with SO_NET_SERVICE_TYPE.

macOS does not support `SO_PRIORITY` on sockets, but it does apparently
support configuring sockets with a priority flag via a parameter called
`SO_NET_SERVICE_TYPE`. It doesn't appear to be especially well
documented, but it seems to exist as far back as 10.11 (El Capitan).

This patch sets QoSSession to treat connections as
"responsive multimedia audio/video", which some docs appear to describe
as "low delay tolerant, low-medium, loss tolerant, elastic flow,
variable packet interval, rate and size".

dwilliams27 committed:
chore: remove unused Logs/Mail/ dir

Sam-Belliveau committed:
Prevent profiles with no device specification from resetting device

Sam-Belliveau committed:
Add SDL Gamepad stock profile

mspeder committed:
RealtekFirmwareLoader: Alternate USB PID for Ugreen CM390

cscd98 committed:
Update libusb submodule to v1.0.29

JMC47 committed:
Merge pull request #13634 from jordan-woyak/enum-concepts

Common: Replace enable_if with concepts and other minor modernizations.

JMC47 committed:
Merge pull request #13882 from jordan-woyak/sdl-win32-msg

SDL: Remove old Windows joystick bug workarounds.

JMC47 committed:
Merge pull request #13884 from jordan-woyak/wiimote-report-timing

WiimoteReal: Send reports with proper timing for theoretically better speaker data.

cscd98 committed:
Update hidapi submodule to v0.15.0

cscd98 committed:
Use hidapi-src/windows on MINGW as well as MSVC

JMC47 committed:
Merge pull request #13897 from TryTwo/Imgui_Font_Size

Qt/Imgui: Add SpinBox to adjust imgui font size.

JMC47 committed:
Merge pull request #13940 from iwubcode/glslang_update

Externals / VideoCommon: update glslang to latest, turn into submodule

JMC47 committed:
Merge pull request #13945 from iwubcode/math_transpose_function

Common: add transpose function to Matrix class

JMC47 committed:
Merge pull request #13951 from TellowKrinkle/InterpreterOpt

Cached Interpreter: Speculative devirtualization

JMC47 committed:
Merge pull request #13954 from TryTwo/debug_crash_fix

Crash fix: Add debug check to JIT Profiling bool.

JMC47 committed:
Merge pull request #13968 from dwilliams27/remove-logs-mail

Logging: Remove unused Logs/Mail/ dir

JMC47 committed:
Merge pull request #13969 from mspeder/ugreen

RealtekFirmwareLoader: Alternate USB PID for Ugreen CM390

JMC47 committed:
Merge pull request #13961 from Sam-Belliveau/sdl-gamepad-profile

Add SDL Gamepad Profile

JMC47 committed:
Merge pull request #13915 from JordanTheToaster/Quiche

Externals: Update gtest to commit eb2d85e

JoshuaVandaele committed:
Externals: Update glslang to 16.0.0 and allow using system glslang

jordan-woyak committed:
SDL: Name the hotplug thread.

jordan-woyak committed:
SDL: Disable DirectInput handling to work around hangs with the "8BitDo Ultimate 2" controller.

cscd98 committed:
Lower case windows includes for case sensitive file systems

JMC47 committed:
Merge pull request #13909 from Dentomologist/codewidgets_show_code_approval_in_hardcore_mode

CodeWidgets: Show code approval in Hardcore mode

JMC47 committed:
Merge pull request #13245 from JosJuice/mbp-condition-flush

Jit: Flush registers used in memory breakpoint conditions

cscd98 committed:
Fix various comptability issues with mingw

JMC47 committed:
Merge pull request #13161 from oltolm/fix_qt6_warnings

dolphinQt: fix Qt6 deprecation warnings

JMC47 committed:
Merge pull request #12329 from Dentomologist/balloontip_fix_premature_close_on_balloontip_hover

BalloonTip: Don't hide when the BalloonTip blocks the cursor

JMC47 committed:
Merge pull request #13260 from JosJuice/android-gcadapter-hotplug-callback

Android: Detect GCAdapter hotplug using BroadcastReceiver

JMC47 committed:
Merge pull request #13082 from Dentomologist/generalpane_hardcore_mode_tooltip_clarifications

GeneralPane: Add Hardcore Mode tooltip clarifications

JMC47 committed:
Merge pull request #13974 from JoshuaVandaele/system-glslang

Externals: Update glslang to 16.0.0 and allow using system glslang

mbc07 committed:
RealtekFirmwareLoader: more RTL8761B device IDs

JoshuaVandaele committed:
Externals: Allow building SDL without X11/Wayland

paulo101977 commented:
<img width="2451" height="1359" alt="image" src="https://github.com/user-attachments/assets/aba4cd63-6edd-45c3-9391-9c7c9da07fad" />

-----------------------------------------------------------------------------------------------------------

<img width="1716" height="939" alt="image" src="https://github.com/user-attachments/assets/349e6e2c-d66e-431f-8d22-207e2d74670f" />

paulo101977 commented:
After your changes (on retroarch):

<img width="1491" height="825" alt="image" src="https://github.com/user-attachments/assets/a34eb87c-9886-497c-94f0-eea9b92bffb7" />

<img width="2469" height="1878" alt="image" src="https://github.com/user-attachments/assets/9e7640bd-b83b-4cf0-8d32-b39ec01884c8" />

paulo101977 commented:
Regarding the messages, I think changing the macros in Source\Core\Common\MsgHandler.h will fix the problem.
I will test here.

JosJuice committed:
Merge pull request #13929 from SuperSamus/dcbz-jit-improvements

Jit: Small `dcbz` improvements

Dentomologist committed:
Jit64: Fix dcbz regression

Fix a regression from d1ba84987 that caused freezes and/or
graphical/audio glitches in various games.

jordan-woyak committed:
Merge pull request #13985 from Dentomologist/jit64_fix_dcbz_regression

Jit64: Fix dcbz regression

SuperSamus committed:
dcbz: Fix AVX path

jordan-woyak committed:
Merge pull request #13972 from cscd98/libusb-update

Update libusb submodule to v1.0.29

jordan-woyak committed:
Merge pull request #13684 from JoshuaVandaele/discordrpc-cmake4

discord-rpc: Bump CMake minimum requirement

jordan-woyak committed:
Merge pull request #13982 from jordan-woyak/sdl-disable-dinput

SDL: Disable DirectInput handling to work around hangs on shutdown.

JosJuice committed:
Merge pull request #13986 from SuperSamus/dcbz-avx-fix

dcbz: Fix AVX path

jordan-woyak committed:
Externals: Bump SDL version to release-3.2.24.

jordan-woyak committed:
Merge pull request #13980 from JoshuaVandaele/sdl-noui

Externals: Allow building SDL without X11/Wayland

jordan-woyak committed:
Merge pull request #13926 from Dentomologist/settings_open_file_dialog_at_user_path

Settings: Open file dialogs at user path instead of custom path

JMC47 committed:
Merge pull request #13907 from TryTwo/PR_Cheat_Refresh

Cheats: Refresh active code after editing it.

jordan-woyak committed:
WiimoteReal/IOhidapi: Remove accidentally included HID Profile byte from write test.

jordan-woyak committed:
WiimoteReal/IOhidapi: Move VID/PID check to a helper function that other backends can use in the future.

jordan-woyak committed:
Merge pull request #13955 from Xphalnos/master

Externals: Update FMT to 12.0.0

jordan-woyak committed:
WiimoteReal/IOhidapi: Minor warning fixes and cleanups.

jordan-woyak committed:
WiimoteReal/IOhidapi: Log the currently attached hid driver on Linux.

AdmiralCurtiss committed:
Merge pull request #13987 from jordan-woyak/sdl-version-bump

Externals: Bump SDL version to release-3.2.24.

AdmiralCurtiss committed:
Merge pull request #13946 from JosJuice/i18n-2025-09-17

i18n: Add comments

AdmiralCurtiss committed:
Merge pull request #13752 from jordan-woyak/xlib-duplicate-key-names

ControllerInterface/Xlib: Combine keycodes with the same name to fix non-working inputs.

AdmiralCurtiss committed:
Merge pull request #13914 from dreamsyntax/discord-rpc-update

Externals: Update discord-rpc

jordan-woyak committed:
WiimoteReal/IOLinux: Increase inquiry length to 3.84 seconds. This is what Wii games use and it seems to work better.

jeapostrophe commented:
Thank you @cscd98; great work! I've tested on Windows and it runs great for me.

AdmiralCurtiss committed:
Merge pull request #13957 from jordan-woyak/wmreal-iolinux-fixes

WiimoteReal/IOLinux: Improvements, fixes, and code cleanups.

AdmiralCurtiss committed:
Merge pull request #13871 from Dentomologist/cheatsearchwidget_small_refactoring

CheatSearchWidget: Minor refactoring

SuperSamus committed:
Jit64: Dynamic length of regular jump instruction (for known addresses)
Conditional jumps already do that, so let's be consistent.

cscd98 committed:
iOS and tvOS compatibility changes

Dentomologist committed:
Visual Studio: Fix DolphinNoGUI build

jordan-woyak committed:
Merge pull request #13994 from Dentomologist/visual_studio_fix_dolphinnogui_build

Visual Studio: Fix DolphinNoGUI build

Dentomologist committed:
DeleteDirRecursively: Don't report error for absent directory

Check if the return value of std::filesystem::remove_all is -1 rather
than 0; the former is the specified return value if there's an error
while 0 just means the directory already didn't exist (which is the end
result we want).

Previously error messages such as the following were possible:
E[COMMON]: DeleteDirRecursively: [path]/User/RedirectSession/ failed The
operation completed successfully.

Also adds a period in the error string to make it look nicer.

replicacoil commented:
Thanks to everyone here undertaking this huge effort. While I do see alot of improvements, there is something that broke in comparison to the normal dolphin core namely the ingame audio of Resident Evil 2 and 3 stopped working again. This was a large issue a few years back and I presume the fix to be backported to Retroarch then. This was the patch in the standalone dolphin emulator https://github.com/dolphin-emu/dolphin/commit/f3b8a985e77e0d5157bc08708a470fd8394bb818 and these work on the normal core. But when I use this updated core, the audio issue is back and it stutters.

The stuttering audio issues that were there earlier are fixed for as far as I can tell by testing Super Mario Sunshine and Super Smash Bros. Melee.

replicacoil commented:
I do not see the yellow message in either the normal core or the new one for loading a patch, but renaming the GLEE08.ini to GLEE08.bak screws it up in the normal one so it is the correct one and I do get a message when a save is being written so I don't know why I am not getting that message. 
For easy testing, get Resident Evil 3 (USA). When on the main menu you have 3 options: New Game, Load Game and Options. Selecting options has a bgm, this screws up if the patch is not applied in the normal core.

paulo101977 commented:
@cscd98 Last commits dont fix the problem. I'll see if I can find some time to look into it this week.

<img width="3102" height="1875" alt="image" src="https://github.com/user-attachments/assets/45833594-c51f-4773-ae5a-4d4686aee6c5" />

JosJuice committed:
Merge pull request #13995 from Dentomologist/deletedirrecursively_dont_report_error_if_directory_is_absent

DeleteDirRecursively: Don't report error for absent directory

replicacoil commented:
I replaced the function: `static std::string CreateSysDirectoryPath()` on line 755 in the file: FileUtil.cpp to this:

```
static std::string CreateSysDirectoryPath()
{
  const std::string sys_directory = "D:\\TEST RETRO\\system\\dolphin-emu\\Sys\\";
  INFO_LOG_FMT(COMMON, "CreateSysDirectoryPath: Setting to {}", sys_directory);
  return sys_directory;
}
```

I loaded up Resident Evil 3 (USA) and got the message `1 game patch(es) enabled` instantly. I went to the Options on the main menu and the game was playing the sound like normal. The 2 lines in the log are: `Common\FileUtil.cpp:778 I[COMMON]: CreateSysDirectoryPath: Setting to D:\TEST RETRO\system\dolphin-emu\Sys\` and `Core\ConfigManager.cpp:240 N[CORE]: Active title: Resident Evil 3: Nemesis (GLEE08)`.

So while I do not know why the system folder determination is incorrect atleast for me, on Win 11, it does seem to be the root issue.

paulo101977 commented:
> I replaced the function: `static std::string CreateSysDirectoryPath()` on line 755 in the file: FileUtil.cpp to this:
> 
> ```
> static std::string CreateSysDirectoryPath()
> {
>   const std::string sys_directory = "D:\\TEST RETRO\\system\\dolphin-emu\\Sys\\";
>   INFO_LOG_FMT(COMMON, "CreateSysDirectoryPath: Setting to {}", sys_directory);
>   return sys_directory;
> }
> ```
> 
> I loaded up Resident Evil 3 (USA) and got the message `1 game patch(es) enabled` instantly. I went to the Options on the main menu and the game was playing the sound like normal. The 2 lines in the log are: `Common\FileUtil.cpp:778 I[COMMON]: CreateSysDirectoryPath: Setting to D:\TEST RETRO\system\dolphin-emu\Sys\` and `Core\ConfigManager.cpp:240 N[CORE]: Active title: Resident Evil 3: Nemesis (GLEE08)`.
> 
> So while I do not know why the system folder determination is incorrect atleast for me, on Win 11, it does seem to be the root issue.
> 
> Edit: I changed it again I do believe that this should be the solution:
> 
> `#elif !defined(__LIBRETRO__) && (defined(_WIN32) || defined(LINUX_LOCAL_DEV))` instead of `#elif defined(_WIN32) || defined(LINUX_LOCAL_DEV)` in the function CreateSysDirectoryPath(). As far as I can tell, this if statement simply gets used by the new core because in standalone dolphin, the Sys folder is in the root of the emulator. With this adjustment, it will only get used when there is no libretro path defined and when it has, it will use: `const std::string sys_directory = s_libretro_sys_directory + DIR_SEP;`

With this configuration, could you test it for me? We're having trouble with OpenGL. Could you set OpenGL in your Retroarch and try saving and loading a state to see if the textures are ok?

paulo101977 commented:
> @replicacoil I've applied your patch. Nice spot, saved me the time!!
> 
> @paulo101977 It wont fix the OpenGL "load state" issue as its not specific to Windows. I've not fixed that yet. It requires much debugging unfortunately (of which currently the solution has eluded me)

I think it's better to put an #ifdef _WIN32 (or other flag) and leave it for later. I'm implementing Vulkan in my frontend, even for these specific cases.

fizzlepicklepop committed:
Add toggles for Speedrun timer + Achievement badges visibility

Added some new visibility toggles so players can choose what to show (or hide) during their RetroAchievements runs:

Toggle for the Speedrun Leaderboards timer

Toggle for Achievement Challenge badges

fizzlepicklepop committed:
fixed the lint error

paulo101977 commented:
> @paulo101977 Have you tried d3d11 or vulkan backends instead until the OpenGL issue is fixed?

Yes, everything seems to be fine with Vulkan and D3D. Only OpenGL seems to be having issues. I also noticed something about the joypad. It seems to be inverted on the Wii. But I didn't test it thoroughly. I was just testing the save functionality. My front-end unfortunately only runs OpenGL, so I'll need to create a Vulkan feature for it.

JMC47 committed:
Merge pull request #13998 from fizzlepicklepop/master

DolphinQt: Add toggles for Speedrun timer + Achievement badges visibility

JMC47 committed:
Merge pull request #13679 from JoshuaVandaele/pugixml-cmake4

pugixml: Update to 1.15 and make it a submodule

JMC47 committed:
Merge pull request #13712 from SuperSamus/cheat-search-improvements

CheatSearchWidget: New feature, writing a value to all selected addresses

JMC47 committed:
Merge pull request #13963 from jordan-woyak/point-passthru-needs-bound-inputs

WiimoteEmu: Require IRPassthrough has any bound inputs to operate.

JMC47 committed:
Merge pull request #13964 from ryanmcgrath/macos/qos

macoS: update QoSSession with SO_NET_SERVICE_TYPE.

JMC47 committed:
Merge pull request #13973 from cscd98/hidapi-update

Hidapi update to v0.15.0

JMC47 committed:
Merge pull request #13979 from mbc07/RealtekFirmwareLoader_newIDs

RealtekFirmwareLoader: more RTL8761B device IDs

JMC47 committed:
Merge pull request #13989 from jordan-woyak/wmreal-hidapi-cleanup

WiimoteReal/IOhidapi: Minor changes / cleanups.

JMC47 committed:
Merge pull request #13996 from SuperSamus/jit64-dynamic-known-jmp-len

Jit64: Dynamic length of regular jump instruction (for known addresses)

cscd98 committed:
Merge upsteam/master into libretro

srk15372 commented:
thank you so much for your hard work.
can't wait to test this thanks
regards

nickkeane commented:
The updated libretro core has a discrepancy being unable to hash RetroAchievements for the RVZ format. ISO is working.
Dolphin Standalone 2509 works with both RVZ and ISO formats.
Core Version: bfac84ab

jvalecillos commented:
Coming back to check this repo after a few years, it is refreshing to know this update finally happened. We were pushing for this 3 years ago at #175, but getting it even considered was quite difficult at the time.
